src/Compilers/CSharp/Portable/Symbols/CollectionArgumentsSignatureOnlyMethodSymbol.cs

```diff
@@ -0,0 +1,152 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// A method symbol that represents a signature for collection expression arguments.
+    /// The compiler is free to emit any equivalent method call or initialization.
+    /// </summary>
+    internal sealed class CollectionArgumentsSignatureOnlyMethodSymbol : MethodSymbol
+    {
+        internal readonly MethodSymbol WellKnownConstructor;
+
+        internal CollectionArgumentsSignatureOnlyMethodSymbol(
+            MethodSymbol wellKnownConstructor,
+            string name,
+            Symbol containingSymbol,
+            ImmutableArray<(string Name, TypeWithAnnotations Type)> parameters,
+            TypeWithAnnotations returnType)
+        {
+            Debug.Assert(wellKnownConstructor is { });
+            Debug.Assert(containingSymbol is { });
+
+            WellKnownConstructor = wellKnownConstructor;
+            Name = name;
+            ContainingSymbol = containingSymbol;
+            var parameterBuilder = ArrayBuilder<ParameterSymbol>.GetInstance(parameters.Length);
+            foreach ((var parameterName, var parameterType) in parameters)
+            {
+                parameterBuilder.Add(SynthesizedParameterSymbol.Create(this, parameterType, parameterBuilder.Count, RefKind.None, parameterName));
+            }
+            Parameters = parameterBuilder.ToImmutableAndFree();
+            ReturnTypeWithAnnotations = returnType;
+        }
+
+        public override string Name { get; }
+
+        public override Symbol ContainingSymbol { get; }
+
+        public override ImmutableArray<TypeParameterSymbol> TypeParameters => [];
+
+        public override ImmutableArray<ParameterSymbol> Parameters { get; }
+
+        public override TypeWithAnnotations ReturnTypeWithAnnotations { get; }
+
+        public override MethodKind MethodKind => MethodKind.Ordinary;
```

Comment:
Should this be `Constructor`?

Reply:
The signature method is not a constructor. It's a factory method with specific parameters, and interface return type.